app/models/spree/invoice.rb

```diff
@@ -0,0 +1,128 @@
+class Spree::Invoice < ActiveRecord::Base
+  belongs_to :invoiceable, polymorphic: true
+  validates :invoiceable, presence: true
+
+  before_save :set_date
+  before_save :set_number, if: :use_sequential_number?
+  after_save :increase_invoice_number, if: :use_sequential_number?
+
+  # Delegates
+
+  delegate :bill_address,
+           :ship_address,
+           :tax_address,
+           :line_items,
+           :display_item_total,
+           :all_adjustments,
+           :shipments,
+           :display_total,
+           :payments, to: :invoiceable
```

Comment:
Yes, because the views need all these methods. So far I see two invoiceable models: `Spree::Reimbursement` and `Spree::Order`. `Spree::Reimbursement` implements many of them, so I'll probably refactor the delegate to `line_items` to something like 
```
def items
  if invoiceable_type == "Spree::Order"
    invoiceable.line_items
  else
    invoiceable.return_items
  end
end 
```